Tests/NautilusTelemetryTests/URLTemplateMatcherTests.swift

```diff
@@ -0,0 +1,204 @@
+// Created by Jon Parise on 6/20/25.
+// Copyright © 2025 Airbnb Inc. All rights reserved.
+
+import Foundation
+import XCTest
+
+@testable import NautilusTelemetry
+
+final class URLTemplateMatcherTests: XCTestCase {
+
+	// MARK: - Path Matching Tests
+
+	func testSimplePathMatching() throws {
+		let matcher = try URLTemplateMatcher(templates: ["/users/{id}"])
+
+		XCTAssertEqual(matcher.match(url: URL(string: "https://example.com/users/123")!), "/users/{id}")
```

Comment:
I realize these URL initializers will never fail, but the `try XCTUnwrap(...)` pattern feels nicer. Perhaps an URL extension that bundles up the pattern?

Reply:
Note that I'm offering two initializers: the throwing one (used here) and a fallible one:

```swift
let matcher = try XCTUnwrap(URLTemplateMatcher(["/users/{id}"]))
```

I think `XCTUnwrap` only makes sense if I switch to that second one in the tests?

Reply:
I was just referring to the force unwraps on URL initialization: `URL(string: "https://example.com/users/123")!`

Reply:
Oh actually, this iOS 16 API might be nice for tests, since it throws and lets you set defaults: 

https://developer.apple.com/documentation/foundation/url/parsestrategy